### DISCUSSION TITLE
How to configure Time warp for indirect relationships on custom objects. Ex: Account should pull Tasks associated to lead and Referral.

### DISCUSSION THREAD

**Discussion by vinodreddys** [deejay-hub/timeline-lwc#231] [Q&A] [answered]
How to configure Time warp for indirect relationships on custom objects. Ex: Account should pull Tasks associated to lead and Referral.

**Comment by deejay-hub** [comment]
Hi @vinodreddys. Can you see the records you need on the standard activity timeline when added to the Account object? We aim to support indirect relationships on the standard Account object via OpenActivity and ActivityHistory relationships. Could you give us steps to replicate your scenario? Are you converting a standard Lead into an Account and Contact?

**Reply by vinodreddys** [reply]
**Scenario:** I want to display all open leads on account object in time warp component. Consider the relationships on the object as Account (Parent) ==> Referral (Child to Account) ==> Leads (Child to Referral, consider this object has no direct relationship with account)
If I create the leads at Referral object level, will I able to see these leads in time warp on account object. Can we pull the lead which have indirect relationships with account in time warp component?

**Reply by deejay-hub** [reply]
Thanks. That makes sense. Time Warp doesn't support this use case. Traversing relationships to the grand child level creates some challenges to get a query to scale and be performant. Your best bet would be to take the code and add your use case onto it. Finding a generic way to do this for any Parent - Child - Grandchild relationship in a way that is fast has eluded me.

**Reply by vinodreddys** [reply]
Got it. Thank you for the quick response.

**Comment by deejay-hub** [comment, marked as answer]
Time Warp doesn't support this use case. Traversing relationships to the grand child level creates some challenges to get a query to scale and be performant. Your best bet would be to take the code and add your use case onto it. Finding a generic way to do this for any Parent - Child - Grandchild relationship in a way that is fast has eluded me.
